src/NuGet.Core/NuGet.Configuration/PublicAPI.Unshipped.txt

```diff
@@ -1,4 +1,4 @@
 #nullable enable
 NuGet.Configuration.PackageSourceMappingProvider.ShouldSkipSave.get -> bool
-~NuGet.Configuration.PackageSourceMapping.SearchForPattern(string packageId) -> string
+NuGet.Configuration.PackageSourceMapping.SearchForPattern(string! packageId) -> string?
```

Comment:
Is `!` valid in the search for pattern? As a public API, I think it makes more sense for it to be null safe instead.

Reply:
I'm not sure how it wouldn't be valid? I believe the fixit analyzer put `!` there.

Existing behavior throws an intentional error message:
https://github.com/NuGet/NuGet.Client/blob/1fbe83fe674911a7f1289dad525be6ff73b7d2e5/src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/SearchTree.cs#L110

Reply:
I missed the fact that this is an intentional behavior.